compiler/rustc_codegen_ssa/src/mir/rvalue.rs

```diff
@@ -119,33 +117,13 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
                     false
                 };
 
-                trace!(?cg_elem.val);
                 match cg_elem.val {
                     OperandValue::Immediate(v) => {
                         if try_init_all_same(bx, v) {
                             return;
                         }
                     }
-                    OperandValue::Pair(a, b) => {
```

Comment:
For future reference, on line 104:

The `try_init_all_same` tries to access a value that should be size u128, but since scalar pairs can be *twice* the size of `u128` it ends up indexing out of bounds.